doc/guides/west/manifest.rst

```diff
@@ -1179,6 +1189,433 @@ processed in this order:
 - If the value is a sequence, its elements are recursively imported in the
   order they appear.
 
+.. _west-manifest-groups:
+
+Project Groups and Active Projects
+**********************************
+
+You can use the ``groups`` keys briefly described :ref:`above
+<west-manifest-files>` to place projects into groups. The ``groups`` key can be
+a project or top level manifest attribute, like this:
+
+.. code-block:: yaml
+
+   manifest:
+     projects:
+       - name: some-project
+         groups: ...
+     groups: ...
+
+You can allow or block project groups. Projects whose groups are all blocked
+are *inactive*; west essentially ignores inactive projects unless explicitly
+requested not to.
+
+Project Groups
+==============
+
+Inside ``manifest: projects:``, you can add a project to one or more groups.
+The ``groups`` key in this case is a list of group names. Group names are
+strings.
+
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+  manifest:
+    projects:
+      - name: project-1
+        groups:
+          - groupA
+      - name: project-2
+        groups:
+          - groupB
+          - groupC
+      - name: project-3
+
+The projects are in these groups:
+
+- ``project-1``: one group, named ``groupA``
+- ``project-2``: two groups, named ``groupB`` and ``groupC``
+- ``project-3``: no groups
+
+Project group names must not contain commas (,), colons (:), or whitespace.
+
+Group names must not begin with a dash (-), but they may contain dashes
+elsewhere in their names: ``foo-bar`` is a valid group name, but ``-foo-bar``
+is not.
+
+Group names are otherwise arbitrary strings. Group names are case sensitive.
+
+As a restriction, no project may use both ``import:`` and ``groups:``. (This
+avoids some edge cases whose semantics are difficult to specify.)
+
+Allowed and Blocked Project Groups
+==================================
+
+You can allow or block project groups in both your manifest file and
+:ref:`west-config`.
+
+All groups are allowed by default.
+
+Within the manifest file, the top level ``manifest: groups:`` value can be used
+to block project groups. To block a group, prefix its name with a dash (-).
+For example, in this manifest fragment:
+
+.. code-block:: yaml
+
+   manifest:
+     groups: [-groupA,-groupB]
```

Comment:
I think this syntax should support `[-groupA,+groupB,-groupC]` even if `+groupB` is a no-op. This would make it consistent with `manifest.groups` below, possibly make some scripting easier and also make the key name less strange because this `groups` is actually a list of `blocked_groups` in disguise for now.